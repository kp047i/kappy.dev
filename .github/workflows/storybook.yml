name: Storybook Tests

on:
  push:
  pull_request:

jobs:
  storybook:
    name: Storybook Tests
    runs-on: ubuntu-latest
    timeout-minutes: 60
    env:
      PLAYWRIGHT_BROWSERS_PATH: 0
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: 22.13.0
      - name: Enable Corepack
        run: corepack enable
      - name: Install pnpm
        run: |
          corepack prepare pnpm@8.15.5 --activate
          pnpm --version
          which pnpm
      - name: Get pnpm store directory
        shell: bash
        run: |
          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
      - name: Setup pnpm cache
        uses: actions/cache@v4
        with:
          path: ${{ env.STORE_PATH }}
          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
          restore-keys: |
            ${{ runner.os }}-pnpm-store-
      - name: Install dependencies
        run: pnpm install --frozen-lockfile
      - name: Install Playwright browsers
        run: pnpm exec playwright install --with-deps chromium chromium-headless-shell
      - name: Build Storybook
        run: pnpm build-storybook --quiet
      - name: Storybook Vitest tests
        env:
          CI: true
        run: pnpm run test:storybook
