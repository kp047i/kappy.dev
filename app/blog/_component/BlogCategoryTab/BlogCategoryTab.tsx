import Link from "next/link";

import { CATEGORIES } from "../../_const/categories";

export function BlogCategoryTab({
  selectedCategory,
}: Readonly<{
  selectedCategory: string;
}>) {
  return (
    <nav className="flex justify-center space-x-4">
      <Link
        href="/blog"
        className={`${
          selectedCategory === ""
            ? "text-primary-700"
            : "text-secondary-950 opacity-60 hover:opacity-100"
        }`}
      >
        All
      </Link>
      {CATEGORIES.map((category) => (
        <Link
          key={category.key}
          href={`/blog?category=${category.key}`}
          className={`${
            selectedCategory === category.key
              ? "text-primary-700"
              : "text-secondary-950 opacity-60 hover:opacity-100"
          }`}
        >
          {category.label}
        </Link>
      ))}
    </nav>
  );
}
