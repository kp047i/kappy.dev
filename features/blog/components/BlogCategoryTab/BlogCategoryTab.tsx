import clsx from "clsx";
import Link from "next/link";

import { CATEGORIES } from "../../const/categories";

export function BlogCategoryTab({
  selectedCategory,
}: Readonly<{
  selectedCategory: string;
}>) {
  return (
    <nav className="flex justify-center space-x-4">
      <Link
        href="/blog"
        className={clsx(
          "text-sm font-medium transition-colors",
          selectedCategory === ""
            ? "text-primary-700 dark:text-primary-200"
            : "text-secondary-950 opacity-60 hover:opacity-100 dark:text-base-100 dark:hover:text-base-50"
        )}
      >
        All
      </Link>
      {CATEGORIES.map((category) => (
        <Link
          key={category.key}
          href={`/blog?category=${category.key}`}
          className={clsx(
            "text-sm font-medium transition-colors",
            selectedCategory === category.key
              ? "text-primary-700 dark:text-primary-200"
              : "text-secondary-950 opacity-60 hover:opacity-100 dark:text-base-100 dark:hover:text-base-50"
          )}
        >
          {category.label}
        </Link>
      ))}
    </nav>
  );
}
