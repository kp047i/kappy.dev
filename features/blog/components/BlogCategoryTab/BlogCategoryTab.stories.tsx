import { Meta, StoryObj } from "@storybook/nextjs-vite";
import { expect, within } from "storybook/test";

import { BlogCategoryTab } from "./BlogCategoryTab";

const meta: Meta<typeof BlogCategoryTab> = {
  component: BlogCategoryTab,
};

export default meta;
type Story = StoryObj<typeof BlogCategoryTab>;

const defaultArgs = {
  selectedCategory: "tech",
  selectedTag: "",
};

export const Default: Story = {
  args: defaultArgs,
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);

    const allLink = canvas.getByText("All");
    expect(allLink).toHaveAttribute("href", "/blog");

    const techLink = canvas.getByText("Tech");
    expect(techLink).toHaveAttribute("href", "/blog?category=tech");

    const lifeLink = canvas.getByText("Life");
    expect(lifeLink).toHaveAttribute("href", "/blog?category=life");

    const otherLink = canvas.getByText("Other");
    expect(otherLink).toHaveAttribute("href", "/blog?category=other");
  },
};

export const WithSelectedTag: Story = {
  args: {
    selectedCategory: "tech",
    selectedTag: "react",
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);

    const allLink = canvas.getByText("All");
    expect(allLink).toHaveAttribute("href", "/blog?tag=react");

    const techLink = canvas.getByText("Tech");
    expect(techLink).toHaveAttribute("href", "/blog?category=tech&tag=react");
  },
};
