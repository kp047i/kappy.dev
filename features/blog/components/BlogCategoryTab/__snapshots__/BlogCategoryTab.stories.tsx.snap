// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog/components/BlogCategoryTab Default play-test 1`] = `
<nav class="flex justify-center space-x-4">
  <a class="text-sm font-medium transition-colors text-secondary-950 opacity-60 hover:opacity-100 dark:text-base-100 dark:hover:text-base-50"
     href="/blog"
  >
    All
  </a>
  <a class="text-sm font-medium transition-colors text-primary-700 dark:text-primary-200"
     href="/blog?category=tech"
  >
    Tech
  </a>
  <a class="text-sm font-medium transition-colors text-secondary-950 opacity-60 hover:opacity-100 dark:text-base-100 dark:hover:text-base-50"
     href="/blog?category=life"
  >
    Life
  </a>
  <a class="text-sm font-medium transition-colors text-secondary-950 opacity-60 hover:opacity-100 dark:text-base-100 dark:hover:text-base-50"
     href="/blog?category=other"
  >
    Other
  </a>
</nav>
`;
